File: LeetCode/LeetCode Practice Problems/1047- Remove All Adjacent Duplicates In String.cpp
class Solution {
public:
    string removeDuplicates(string S) {
        for(int i=0;i<S.length();) {
            if(i+1<S.length() && S[i] == S[i+1]) {
                S = S.substr(0,i) + S.substr(i+2);
                i=0;
            } else {
                i++;
            }
        }
        
        return S;
    }
};

File: LeetCode/LeetCode Practice Problems/1047-Remove All Adjacent Duplicates In String.java
class Solution {
    public String removeDuplicates(String s) {
        int i = 0 , n = s.length();
        char res[] = s.toCharArray();
        
        for(int j=0;j<n;j++,i++) {
            res[i] = res[j];
            if(i > 0 && res[i] == res[i-1]) 
                i -= 2;
        }
        
        return new String(res, 0,i);
    }
}
